src/lib/ZoomControls.js

```diff
@@ -0,0 +1,134 @@
+import isFinite from 'lodash/isFinite';
+import EventEmitter from 'events';
+import { ICON_ZOOM_IN, ICON_ZOOM_OUT } from './icons/icons';
+
+const CLASS_ZOOM_CURRENT_SCALE = 'bp-zoom-current-scale';
+const CLASS_ZOOM_IN_BUTTON = 'bp-zoom-in-btn';
+const CLASS_ZOOM_OUT_BUTTON = 'bp-zoom-out-btn';
+
+class ZoomControls extends EventEmitter {
+    /** @property {Controls} - Controls object */
+    controls;
+
+    /** @property {HTMLElement} - Controls element */
+    controlsElement;
+
+    /** @property {number} - Current zoom scale */
+    currentScale;
+
+    /** @property {HTMLElement} - Current scale element */
+    currentScaleElement;
+
+    /** @property {number} - Max zoom scale */
+    maxZoom;
+
+    /** @property {number} - Min zoom scale */
+    minZoom;
+
+    /**
+     * [constructor]
+     *
+     * @param {HTMLElement} controls - Viewer controls
+     * @return {Controls} Instance of controls
+     */
+    constructor(controls) {
+        super();
+
+        this.controls = controls;
+        this.controlsElement = controls.controlsEl;
+
+        this.handleZoomIn = this.handleZoomIn.bind(this);
+        this.handleZoomOut = this.handleZoomOut.bind(this);
+    }
+
+    /**
+     * Add the zoom controls
+     *
+     * @param {number} currentScale - Initial scale value, assumes range of 0-1
+     * @param {number} [options.maxZoom] - Maximum zoom, range 0-1
+     * @param {number} [options.minZoom] - Minimum zoom, range 0-1
+     * @param {String} [options.zoomInClassName] - Class name for zoom in button
+     * @param {String} [options.zoomOutClassName] - Class name for zoom out button
+     * @return {void}
+     */
+    add(
+        currentScale,
+        { zoomOutClassName = '', zoomInClassName = '', minZoom = 0, maxZoom = Number.POSITIVE_INFINITY } = {},
+    ) {
+        this.maxZoom = Math.ceil(maxZoom * 100);
+        this.minZoom = Math.ceil(minZoom * 100);
+
+        this.controls.add(
+            __('zoom_out'),
+            this.handleZoomOut,
+            `${CLASS_ZOOM_OUT_BUTTON} ${zoomOutClassName}`,
+            ICON_ZOOM_OUT,
+        );
+        this.controls.add(
+            __('zoom_current_scale'),
+            undefined,
```

Comment:
Will this not reset the scale to 100% or some known factor?

Reply:
Not for now, the decision is to let it be read only for now and then we can iterate on it